packages/core/src/shared/vscode/env.ts

```diff
@@ -277,8 +277,14 @@ export function getApplicationSupportFolder() {
         case 'darwin': {
             return path.join(os.homedir(), 'Library/Application Support')
         }
+        case 'win32': {
+            return process.env.APPDATA
+        }
+        case 'linux': {
+            return process.env.XDG_CACHE_HOME || path.join(os.homedir(), '.cache')
```

Comment:
Should use `fs.getUserHomeDir`, it handles some edge cases. And to avoid the circular dependency, `getApplicationSupportFolder` should live in `fs.ts` (and that makes sense anyways, because this is a filesystem decision).

It also seems like `getApplicationSupportFolder` is getting a cache dir, so maybe `getCacheDir` is a good name.

Reply:
`tryGetFilepathEnvVar` may also be worth looking at in the future. https://github.com/aws/aws-toolkit-vscode/blob/02f6d0bf817253f8eaf1ad6f13c333d2562f0a9c/packages/core/src/shared/fs/fs.ts#L517